src/clj/fluree/db/query/fql/parse.cljc

```diff
@@ -75,9 +75,10 @@
 (defn match-value-binding-map
   [var-match binding-map context]
   (let [attrs (expand-keys binding-map context)
-        val   (get attrs const/iri-value)]
+        {val const/iri-value} attrs]
     (if-let [dt-iri (get-expanded-datatype attrs context)]
-      (if (= const/iri-anyURI dt-iri)
+      (if (or (= const/iri-anyURI dt-iri)
```

Comment:
We should eventually remove this, but I'm ok with keeping it in for now in case someone is using it.